### ISSUE TITLE
WarpDrive 1.12.2-1.5.10 requires Forge 14.23.5.2817+

### ISSUE THREAD

**Issue by oakenmoon** [LemADEC/WarpDrive#406]
**Describe the bug**
It gives a NoClassDefFoud exception when I try to load an existing world or create a new world. Once it happened while trying to load the pack, but I haven't been able to reproduce that crash, though I'm pretty sure that was without the Thermal suite. Same error happens if I add IC2, I can add that crash log too if needed.
Note that I had Dynamic Sword Skills 6.0.1 and SwordSkillsAPI 1.1.0 at the beginning of testing, the crash log was after I removed those to see if it's a conflict. Still happens without the Thermal set, and with Thermal but without CC.

The game crashed whilst exception in server tick loop
Error: java.lang.NoClassDefFoundError: net/minecraftforge/event/world/BlockEvent$EntityPlaceEvent

**Logs**
Link to pastebin/gist or equivalent logs
```
---- Minecraft Crash Report ----

WARNING: coremods are present:
  WarpDriveCore (WarpDrive-1.12.2-1.5.10.jar)
Contact their authors BEFORE contacting forge

// You should try our sister game, Minceraft!

Time: 10/16/19 1:54 PM
Description: Exception in server tick loop

java.lang.NoClassDefFoundError: net/minecraftforge/event/world/BlockEvent$EntityPlaceEvent
	at cr0s.warpdrive.event.WorldHandler.onBlockEvent(WorldHandler.java:149)
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler_142_WorldHandler_onBlockEvent_BlockEvent.invoke(.dynamic)
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:90)
	at net.minecraftforge.fml.common.eventhandler.EventBus.post(EventBus.java:182)
	at net.minecraftforge.event.ForgeEventFactory.onNeighborNotify(ForgeEventFactory.java:162)
	at net.minecraft.world.World.func_175685_c(World.java:489)
	at net.minecraft.world.World.func_175722_b(World.java:440)
	at net.minecraft.world.World.markAndNotifyBlock(World.java:381)
	at net.minecraft.world.World.func_180501_a(World.java:361)
	at net.minecraft.block.BlockDynamicLiquid.func_176375_a(BlockDynamicLiquid.java:175)
	at net.minecraft.block.BlockDynamicLiquid.func_180650_b(BlockDynamicLiquid.java:153)
	at net.minecraft.world.WorldServer.func_175654_a(WorldServer.java:551)
	at net.minecraft.world.WorldServer.func_175684_a(WorldServer.java:531)
	at net.minecraft.block.BlockDynamicLiquid.func_176213_c(BlockDynamicLiquid.java:306)
	at net.minecraft.world.chunk.Chunk.func_177436_a(Chunk.java:614)
	at net.minecraft.world.World.func_180501_a(World.java:343)
	at net.minecraft.block.BlockDynamicLiquid.func_176375_a(BlockDynamicLiquid.java:175)
	at net.minecraft.block.BlockDynamicLiquid.func_180650_b(BlockDynamicLiquid.java:153)
	at net.minecraft.world.WorldServer.func_175654_a(WorldServer.java:551)
	at net.minecraft.world.WorldServer.func_175684_a(WorldServer.java:531)
	at net.minecraft.block.BlockDynamicLiquid.func_176213_c(BlockDynamicLiquid.java:306)
	at net.minecraft.world.chunk.Chunk.func_177436_a(Chunk.java:614)
	at net.minecraft.world.World.func_180501_a(World.java:343)
	at net.minecraft.block.BlockDynamicLiquid.func_176375_a(BlockDynamicLiquid.java:175)
	at net.minecraft.block.BlockDynamicLiquid.func_180650_b(BlockDynamicLiquid.java:153)
	at net.minecraft.world.WorldServer.func_175654_a(WorldServer.java:551)
	at net.minecraft.world.WorldServer.func_175684_a(WorldServer.java:531)
	at net.minecraft.block.BlockDynamicLiquid.func_176213_c(BlockDynamicLiquid.java:306)
	at net.minecraft.world.chunk.Chunk.func_177436_a(Chunk.java:614)
	at net.minecraft.world.World.func_180501_a(World.java:343)
	at net.minecraft.block.BlockDynamicLiquid.func_176375_a(BlockDynamicLiquid.java:175)
	at net.minecraft.block.BlockDynamicLiquid.func_180650_b(BlockDynamicLiquid.java:133)
	at net.minecraft.world.WorldServer.func_175654_a(WorldServer.java:551)
	at net.minecraft.world.WorldServer.func_175684_a(WorldServer.java:531)
	at net.minecraft.block.BlockDynamicLiquid.func_176213_c(BlockDynamicLiquid.java:306)
	at net.minecraft.world.chunk.Chunk.func_177436_a(Chunk.java:614)
	at net.minecraft.world.World.func_180501_a(World.java:343)
	at net.minecraft.block.BlockDynamicLiquid.func_176375_a(BlockDynamicLiquid.java:175)
	at net.minecraft.block.BlockDynamicLiquid.func_180650_b(BlockDynamicLiquid.java:153)
	at net.minecraft.world.WorldServer.func_175654_a(WorldServer.java:551)
	at net.minecraft.world.WorldServer.func_175684_a(WorldServer.java:531)
	at net.minecraft.block.BlockDynamicLiquid.func_176213_c(BlockDynamicLiquid.java:306)
	at net.minecraft.world.chunk.Chunk.func_177436_a(Chunk.java:614)
	at net.minecraft.world.World.func_180501_a(World.java:343)
	at net.minecraft.block.BlockDynamicLiquid.func_176375_a(BlockDynamicLiquid.java:175)
	at net.minecraft.block.BlockDynamicLiquid.func_180650_b(BlockDynamicLiquid.java:153)
	at net.minecraft.world.WorldServer.func_175654_a(WorldServer.java:551)
	at net.minecraft.world.WorldServer.func_175684_a(WorldServer.java:531)
	at net.minecraft.block.BlockDynamicLiquid.func_176213_c(BlockDynamicLiquid.java:306)
	at net.minecraft.world.chunk.Chunk.func_177436_a(Chunk.java:614)
	at net.minecraft.world.World.func_180501_a(World.java:343)
	at net.minecraft.block.BlockDynamicLiquid.func_176375_a(BlockDynamicLiquid.java:175)
	at net.minecraft.block.BlockDynamicLiquid.func_180650_b(BlockDynamicLiquid.java:133)
	at net.minecraft.world.WorldServer.func_175654_a(WorldServer.java:551)
	at net.minecraft.world.WorldServer.func_175684_a(WorldServer.java:531)
	at net.minecraft.block.BlockDynamicLiquid.func_176213_c(BlockDynamicLiquid.java:306)
	at net.minecraft.world.chunk.Chunk.func_177436_a(Chunk.java:614)
	at net.minecraft.world.World.func_180501_a(World.java:343)
	at net.minecraft.block.BlockDynamicLiquid.func_176375_a(BlockDynamicLiquid.java:175)
	at net.minecraft.block.BlockDynamicLiquid.func_180650_b(BlockDynamicLiquid.java:153)
	at net.minecraft.world.WorldServer.func_175654_a(WorldServer.java:551)
	at net.minecraft.world.WorldServer.func_175684_a(WorldServer.java:531)
	at net.minecraft.block.BlockDynamicLiquid.func_176213_c(BlockDynamicLiquid.java:306)
	at net.minecraft.world.chunk.Chunk.func_177436_a(Chunk.java:614)
	at net.minecraft.world.World.func_180501_a(World.java:343)
	at net.minecraft.block.BlockDynamicLiquid.func_176375_a(BlockDynamicLiquid.java:175)
	at net.minecraft.block.BlockDynamicLiquid.func_180650_b(BlockDynamicLiquid.java:153)
	at net.minecraft.world.World.func_189507_a(World.java:2677)
	at net.minecraft.world.gen.feature.WorldGenLiquids.func_180709_b(WorldGenLiquids.java:88)
	at net.minecraft.world.biome.BiomeDecorator.func_150513_a(BiomeDecorator.java:352)
	at net.minecraft.world.biome.BiomeDecorator.func_180292_a(BiomeDecorator.java:88)
	at net.minecraft.world.biome.Biome.func_180624_a(Biome.java:225)
	at net.minecraft.world.gen.ChunkGeneratorOverworld.func_185931_b(ChunkGeneratorOverworld.java:464)
	at net.minecraft.world.chunk.Chunk.func_186034_a(Chunk.java:1019)
	at net.minecraft.world.chunk.Chunk.func_186030_a(Chunk.java:999)
	at net.minecraft.world.gen.ChunkProviderServer.func_186025_d(ChunkProviderServer.java:157)
	at net.minecraft.server.MinecraftServer.func_71222_d(MinecraftServer.java:344)
	at net.minecraft.server.integrated.IntegratedServer.func_71247_a(IntegratedServer.java:142)
	at net.minecraft.server.integrated.IntegratedServer.func_71197_b(IntegratedServer.java:156)
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:486)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.ClassNotFoundException: net.minecraftforge.event.world.BlockEvent$EntityPlaceEvent
	at net.minecraft.launchwrapper.LaunchClassLoader.findClass(LaunchClassLoader.java:191)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 81 more
Caused by: java.lang.NullPointerException


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- System Details --
Details:
	Minecraft Version: 1.12.2
	Operating System: Windows 10 (amd64) version 10.0
	Java Version: 1.8.0_51, Oracle Corporation
	Java VM Version: Java HotSpot(TM) 64-Bit Server VM (mixed mode), Oracle Corporation
	Memory: 645882608 bytes (615 MB) / 1174405120 bytes (1120 MB) up to 2147483648 bytes (2048 MB)
	JVM Flags: 8 total; -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xmx2G -XX:+UnlockExperimentalVMOptions -XX:+UseG1GC -XX:G1NewSizePercent=20 -XX:G1ReservePercent=20 -XX:MaxGCPauseMillis=50 -XX:G1HeapRegionSize=32M
	IntCache: cache: 1, tcache: 1, allocated: 12, tallocated: 94
	FML: MCP 9.42 Powered by Forge 14.23.5.2816 15 mods loaded, 15 mods active
	States: 'U' = Unloaded 'L' = Loaded 'C' = Constructed 'H' = Pre-initialized 'I' = Initialized 'J' = Post-initialized 'A' = Available 'D' = Disabled 'E' = Errored

	| State  | ID                | Version                | Source                                          | Signature                                |
	|:------ |:----------------- |:---------------------- |:----------------------------------------------- |:---------------------------------------- |
	| LCHIJA | minecraft         | 1.12.2                 | minecraft.jar                                   | None                                     |
	| LCHIJA | mcp               | 9.42                   | minecraft.jar                                   | None                                     |
	| LCHIJA | FML               | 8.0.99.99              | forge-1.12.2-14.23.5.2816.jar                   | e3c3d50c7c986df74c645c0ac54639741c90a557 |
	| LCHIJA | forge             | 14.23.5.2816           | forge-1.12.2-14.23.5.2816.jar                   | e3c3d50c7c986df74c645c0ac54639741c90a557 |
	| LCHIJA | WarpDriveCore     | 1.12.2-1.5.10.SNAPSHOT | minecraft.jar                                   | None                                     |
	| LCHIJA | cctweaked         | 1.85.0                 | cc-tweaked-1.12.2-1.85.0.jar                    | None                                     |
	| LCHIJA | computercraft     | 1.85.0                 | cc-tweaked-1.12.2-1.85.0.jar                    | None                                     |
	| LCHIJA | codechickenlib    | 3.2.3.357              | CodeChickenLib-1.12.2-3.2.3.357-universal.jar   | f1850c39b2516232a2108a7bd84d1cb5df93b261 |
	| LCHIJA | redstoneflux      | 2.1.0                  | RedstoneFlux-1.12-2.1.0.6-universal.jar         | 8a6abf2cb9e141b866580d369ba6548732eff25f |
	| LCHIJA | cofhcore          | 4.6.3                  | CoFHCore-1.12.2-4.6.3.27-universal.jar          | None                                     |
	| LCHIJA | cofhworld         | 1.3.1                  | CoFHWorld-1.12.2-1.3.1.7-universal.jar          | 8a6abf2cb9e141b866580d369ba6548732eff25f |
	| LCHIJA | thermalfoundation | 2.6.3                  | ThermalFoundation-1.12.2-2.6.3.27-universal.jar | 8a6abf2cb9e141b866580d369ba6548732eff25f |
	| LCHIJA | thermalexpansion  | 5.5.4                  | ThermalExpansion-1.12.2-5.5.4.43-universal.jar  | 8a6abf2cb9e141b866580d369ba6548732eff25f |
	| LCHIJA | thermaldynamics   | 2.5.5                  | ThermalDynamics-1.12.2-2.5.5.21-universal.jar   | 8a6abf2cb9e141b866580d369ba6548732eff25f |
	| LCHIJA | warpdrive         | 1.12.2-1.5.10.SNAPSHOT | WarpDrive-1.12.2-1.5.10.jar                     | f7be6b40743c6a8205df86c5e57547d578605d8a |

	Loaded coremods (and transformers): 
WarpDriveCore (WarpDrive-1.12.2-1.5.10.jar)
  cr0s.warpdrive.core.ClassTransformer
	GL info: ~~ERROR~~ RuntimeException: No OpenGL context found in the current thread.
	Profiler Position: N/A (disabled)
	Player Count: 0 / 8; []
	Type: Integrated Server (map_client.txt)
	Is Modded: Definitely; Client brand changed to 'fml,forge'```

**Comment by LemADEC** [comment]
Your forge version 14.23.5.2816 is way outdated.

**Comment by oakenmoon** [comment]
When I start it on Forge 14.23.5.2768, it claims to need at least 2816. Thanks for the help!
![image](https://user-images.githubusercontent.com/56654749/66986536-57227e00-f07c-11e9-9f38-5aa1d45f25e3.png)

**Comment by LemADEC** [comment]
2816 was needed for another dependency. 2817 requirement was known earlier.
Either way, you should always try latest version to get latest bug fixes :)

**Comment by LemADEC** [comment]
Fixed in a91dac78c8654dcfc90347793a69df35ff809dc0 released under 1.12.2-1.5.11
